### ISSUE TITLE
Onivim is available via AUR, include in installation methods?

### ISSUE THREAD

**Issue by Kudostoy0u** [onivim/oni2#3409]
I am on Garuda linux, fish shell, and I found a different way to install onivim2.

```fish
╭─kudos@kudos in ~ took 40ms
[🔴] × pacman -Q --info onivim2
Name            : onivim2-git
Version         : 2033.41c412d13-1
Description     : Native, lightweight modal code editor
Architecture    : x86_64
URL             : https://github.com/onivim/oni2
Licenses        : custom:OutrunLabsEULA
Groups          : None
Provides        : onivim2
Depends On      : None
Optional Deps   : None
Required By     : None
Optional For    : None
Conflicts With  : onivim2
Replaces        : None
Installed Size  : 349.87 MiB
Packager        : Pedro Henrique Lara Campos <root@pedrohlc.com>
Build Date      : Sun 11 Apr 2021 12:37:31 PM CDT
Install Date    : Tue 13 Apr 2021 03:26:18 PM CDT
Install Reason  : Explicitly installed
Install Script  : Yes
Validated By    : Signature


╭─kudos@kudos in ~ took 42ms
╰─λ pamac info onivim2
Name                  : onivim2
Version               : 0.5.4-1
Description           : Native, lightweight modal code editor
URL                   : https://github.com/onivim/oni2
Licenses              : custom:OutrunLabsEULA
Repository            : AUR
Make Dependencies     : esy ragel nodejs wget bzip2 fontconfig fuse2 glu gtk3 harfbuzz libglvnd libice libpng
libsm libx11 libxcursor libxext libxi libxinerama libxrandr libxt libxxf86vm m4 nasm
python2 clang node-gyp
Maintainer            : freswa
First Submitted       : 01/04/2020
Last Modified         : 03/03/2021
Votes                 : 2


╭─kudos@kudos in ~ took 729ms
╰─λ
```
Here is onivim2 on the AUR: https://aur.archlinux.org/packages/onivim2/

It's much more convenient to install with pacman's prebuilt binaries and fast mirrors rather than building it from source with the help of `git clone`, which the download speed is ~250KiB/s for me.

I don't see this as a way to install onivim [in the docs](https://onivim.github.io/docs/getting-started/installation), and it's under the same license, so I think it would be convenient for arch/arch-based distro users if you could perhaps include this as a way to install onivim2?

Thanks, 
Kudos

**Comment by jerabaul29** [comment]
This is a package built from the latest code available on github, right? Curious of how "compatible with the current license" distributing such binary can be?

**Comment by kinglouie** [comment]
@Kudostoy0u I am wondering if your linux distro uses the official arch repos or do they maintain their own? In the official arch repos there is no onivim2 prebuilt package and the packages on the aur are not distributing prebuilt binaries only a PKGBUILD file which lets you compile the binary yourself. the `onivim2-git` package is building from master and  the `onivim2` package is pinned to specific release. 
For me a prebuilt package would be really nice because everytime I want to checkout the latest master changes I install the onivim2-git package and then my machine is unusable for a good 30 minutes while building.

**Comment by Kudostoy0u** [comment]
> This is a package built from the latest code available on github, right? Curious of how "compatible with the current license" distributing such binary can be?

Well, this package is being distributed under the same license according to the package info -> `Licenses              : custom:OutrunLabsEULA`

If you want to see it yourself find it in the AUR -> https://aur.archlinux.org/packages/onivim2/

I'm not sure exactly how it works because I'm pretty new to the Linux community, but how I comprehended the information.

**Comment by jerabaul29** [comment]
What I was thinking is that there is a form a barrier for "mis-using" the license from the download page: you have to either say you are academic / education / nonprofit to get it for free, or you have to pay. The alternative, building yourself, is a form of barrier. With this solution, there is no barrier for violating the license, it becomes a passive process of "I did a package install, it never asked me what I will use it for".

Of course, this is only my 2 cents. Just thinking that the author has to keep a tiny bit of barrier to use to make a bit of money and live of it.

**Comment by Kudostoy0u** [comment]
> For me a prebuilt package would be really nice because everytime I want to checkout the latest master changes I install the onivim2-git package and then my machine is unusable for a good 30 minutes while building.

This is confusing for me, because when I downloaded it using `pacman -S`, it almost instantly installed. Maybe it's a pre-built PKGBUILD, if that's even possible, or somehow got its way into the official repositories.

**Comment by kinglouie** [comment]
@Kudostoy0u then this means that your distro has its own official repos and is not using the arch repos. I don't know about the legal aspects of this but for me it seems a bit fishy to distribute software binaries with a custom licence.

**Comment by gbaranski** [comment]
AUR package doesn't have `unzip` dependency which is required while building.

**Comment by kawaiiamber** [comment]
From the Outrun Labs EULA:
> 1.4 Restrictions on Use.

> You shall use the Software strictly in accordance with the terms of this Agreement and shall not: (a) distribute the Software (b) make the Software available over a network or other environment permitting access or use by multiple users at the same time; (c) violate any applicable laws, rules, or regulations in connection with Your access or use of the Software; (d) use the Software for any revenue generating endeavor, commercial enterprise, employment, or other purpose for which it is not designed or intended; (e) remove proprietary notices from the Software or the Documentation.

Since onivim2 does *NOT* allow distribution of the software according to section 1.4 (a), and the PKGBUILD pulls from this repo and not the MIT repo, it would seem that the AUR is in license violation?

**Comment by Bad3r** [comment]
The AUR package seems to be outdated and [`onivim2-git`](https://aur.archlinux.org/packages/onivim2-git/)  is orphaned and fails to build
```py
/usr/bin/esy: line 2: /opt/esy/node_modules/esy/_build/default/bin/esy.exe: No such file or directory
==> ERROR: A failure occurred in build().
    Aborting...
error making: onivim2-git
```

**Comment by adrielsand** [comment]
@Bad3r have you been able to build the package in any way? I've tried multiple ways and it usually fails

**Comment by pmkap** [comment]
The problem is esy package in the AUR, there is an error in the PKGBUILD. https://github.com/onivim/oni2/issues/3409#issuecomment-855713669 . Check out the latest comment, it describes what you need to fix...

**Comment by adrielsand** [comment]
@gbaranski I added the `unzip` dependency to the PKGBUILD

@pmkap I tried to fix that change in the PKGBUILD but it says

```
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I don't know what to do to fix that

**Comment by pmkap** [comment]
Are you trying to push to the AUR? Just fix the PKGBUILD and run `makepkg -si`.

**Comment by adrielsand** [comment]
@pmkap Yes, trying to push to the AUR like I did with the `onivim2` package. Maybe I don't have necessary permissions?

**Comment by pmkap** [comment]
@adrielsand Yes, only the maintainer of the package can push changes.
The maintainter fixed the esy package. You should be able now to install everything as you usually do. If not, please comment on the AUR package site. This is not an issue of Onivim2 and doesn't really belong here.
